Misc app fixes for 24.2

cnathe opened the pull request:
#### Rationale
- Issue [49481](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49481): Domain designer issue if the field index changes (reordering fields), we need to update the validValues state
- Issue [49439](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49439): filter out inactive users from insert and update form query selects
- Issue [49502](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49502): If the Editable Grid lookup cell is a measurement unit, then we need to use the unit's display value
- Issue [49383](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49383): App error when a folder admin access profile page from a child folder
- Issue [49417](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=49417): App user details modal to show above the app header details popover

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1401
- https://github.com/LabKey/labkey-ui-premium/pull/313
- https://github.com/LabKey/sampleManagement/pull/2407
- https://github.com/LabKey/biologics/pull/2672
- https://github.com/LabKey/inventory/pull/1175

#### Changes
- TextChoiceOptions to reset validValues state on field index change
- LookupCell to use display value for units lookup field
- GroupsList to use app home container for user perm check
- ProfilePage to fail fast when impersonating user
- QueryColumn to add queryFilter for active users in user lookup case

cnathe committed:
Issue 49481: Domain designer issue if the field index changes (reordering fields), we need to update the validValues state

cnathe committed:
3.9.0-fb-misc242can.0

labkey-susanh commented on packages/components/src/internal/components/domainproperties/TextChoiceOptions.tsx:
```diff
@@ -334,6 +334,11 @@ export const TextChoiceOptions: FC<Props> = memo(props => {
+    useEffect(() => {
+        // Issue 49481: if the field index changes (reordering fields), we need to update the validValues state
+        setValidValues(field.textChoiceValidator?.properties.validValues ?? []);
+    }, [field.textChoiceValidator?.properties.validValues, fieldId]);
```
`fieldId` doesn't seem to be used here. Is it intentionally included as a dependency?

cnathe replied:
Yes, that is used as the mechanism for detecting that the field's row index has changed and that the values in state need to be updated.

cnathe committed:
Issue 49439: filter out inactive users from insert and update form query selects

cnathe committed:
missed from prev commit

cnathe committed:
Issue 49502: If the Editable Grid lookup cell is a measurement unit, then we need to use the unit's display value

cnathe committed:
Issue 49383: App error when a folder admin access profile page from a child folder
- use the app home folder for admin perm check before querying for group memebership info
- don't query for API keys if impersonatingUser

cnathe committed:
Merge branch 'develop' into fb_misc242can

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.11.0-fb-misc242can.0

cnathe committed:
Issue 49417: App user details modal to show above the app header details popover

cnathe committed:
Merge branch 'develop' into fb_misc242can

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.11.1-fb-misc242can.0

cnathe committed:
jest test fix for GroupsList.tsx

labkey-susanh commented on packages/components/src/internal/components/editable/LookupCell.tsx:
```diff
@@ -114,6 +114,12 @@ export class LookupCell extends PureComponent<LookupCellProps> {
+        let selectValue = isMultiple ? rawValues : rawValues[0];
+        // Issue 49502: If the lookup is a measurement unit, then we need to use the unit's display value
+        if (col.isUnitsLookup()) {
+            selectValue = values.find(vd => vd.display !== undefined)?.display;
```
Kind of weird that this uses a find. There's only one value possible, right?

cnathe replied:
Yeah...I was just matching what this component is doing for the `rawValues` const. But for the units lookup case, the array would only ever have one element. So I agree that find is a little weird.

labkey-susanh commented on packages/components/src/internal/components/user/ProfilePage.tsx:
```diff
@@ -319,21 +319,21 @@ const APIKeysPanelBody: FC<APIKeysPanelBodyProps & InjectedQueryModels> = props
+export const APIKeysPanel: FC<APIKeysPanelBodyProps> = props => {
+    const { homeContainer, impersonatingUser } = useServerContext();
+    const configs: QueryConfigMap = {};
+    if (!impersonatingUser) {
```
Thanks for fixing this.

labkey-susanh commented on packages/components/src/public/QueryColumn.ts:
```diff
@@ -324,7 +328,7 @@ export class QueryColumn implements IQueryColumn {
 
     isUnitsLookup(): boolean {
-        return this.isLookup && this.lookup.queryName === QueryColumn.MEASUREMENT_UNITS_QUERY;
+        return this.isLookup() && this.lookup.queryName === QueryColumn.MEASUREMENT_UNITS_QUERY;
```
Surprising that this didn't cause more errors to be logged somewhere...

cnathe replied:
I was surprised too...

cnathe committed:
Cleanup props for GroupsList.tsx, unused currentUser

cnathe committed:
3.11.1-fb-misc242can.1

cnathe committed:
add test case for QueryColumn.hasQueryFilters for user lookup

cnathe committed:
Merge branch 'develop' into fb_misc242can

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
npm run lint-branch-fix

cnathe committed:
Update release notes with version number and release date

cnathe committed:
3.13.0
